$OpenBSD: patch-src_http_ngx_http_spdy_c,v 1.1 2014/03/25 02:39:38 william Exp $

Adapt fix for CVE-2014-0088 and apply fix for CVE-2014-0133

Taken from:

http://nginx.org/download/patch.2014.spdy.txt
http://nginx.org/download/patch.2014.spdy2.txt


--- src/http/ngx_http_spdy.c.orig	Tue Nov 19 05:03:48 2013
+++ src/http/ngx_http_spdy.c	Fri Mar 21 22:11:30 2014
@@ -866,7 +866,7 @@ ngx_http_spdy_state_headers(ngx_http_spdy_connection_t
         ngx_log_debug1(NGX_LOG_DEBUG_HTTP, r->connection->log, 0,
                        "spdy headers count: %ui", sc->headers);
 
-        if (ngx_list_init(&r->headers_in.headers, r->pool, sc->headers + 3,
+        if (ngx_list_init(&r->headers_in.headers, r->pool, 20,
                           sizeof(ngx_table_elt_t))
             != NGX_OK)
         {
@@ -1478,7 +1478,7 @@ static u_char *
 ngx_http_spdy_state_save(ngx_http_spdy_connection_t *sc,
     u_char *pos, u_char *end, ngx_http_spdy_handler_pt handler)
 {
-#if (NGX_DEBUG)
+#if 1
     if (end - pos > NGX_SPDY_STATE_BUFFER_SIZE) {
         ngx_log_error(NGX_LOG_ALERT, sc->connection->log, 0,
                       "spdy state buffer overflow: "
